plugins/lua/zone.lua

```diff
@@ -1227,18 +1227,21 @@ local function check_valid_unit()
         and dfhack.units.isFortControlled(unit)
         and dfhack.units.isAlive(unit)
         and dfhack.units.isAnimal(unit)
+        or dfhack.units.isTamable(unit)  -- account for caged animals
```

Comment:
sticking an `or` in the middle of a bunch of `and`s is difficult to read. Could you refactor this into sub-functions? e.g.
```lua
return unit and (isFortAnimal(unit) or isCagedWildAnimal(unit))
```